include/taskomat/exceptions.h

```diff
@@ -57,6 +61,46 @@ class Error : public std::runtime_error
     using std::runtime_error::runtime_error;
 };
 
+/**
+ * An exception class storing the index of the step in which an error occurred, in
+ * addition to the error message.
+ *
+ * \code
+ * try
+ * {
+ *     throw task::ErrorAtIndex("An error has occurred", 42);
+ * }
+ * catch (const task::ErrorAtIndex& e)
+ * {
+ *     std::cerr << e.what() << " in step " << e.get_index() << "\n";
+ * }
+ * \endcode
+ *
+ * \note
+ * task::ErrorAtIndex inherits from std::runtime_error. It can therefore be caught by
+ * `catch (const std::exception&)`, `catch (const std::runtime_error&)`,
+ * `catch (const task::Error&)`, and `catch (const task::ErrorAtIndex&)`.
+ */
+class ErrorAtIndex : public Error
+{
+public:
+    ErrorAtIndex(const std::string& msg, StepIndex index)
+        : Error(msg)
+        , index_(index)
+    {}
+
+    ErrorAtIndex(const char* msg, StepIndex index)
+        : Error(msg)
+        , index_(index)
+    {}
+
+    /// Return the associated step index.
+    StepIndex get_index() const noexcept { return index_; }
+
+private:
+    StepIndex index_ = 0;
```

Comment:
Why has `index_` a default member initializer? I'm not sure about the consequences, but it will be never needed. Maybe just a failsafe? If you look into the `std::runtime_error` you see that the 'what' data member has also no default initializer and its initialization is (only) guaranteed because of the constructors.

Reply:
Yes, it is a failsafe. I just do not like to leave raw integers or floats uninitialized. I'd hope that this gets optimized away as there is basically no code path using the initialization.